xunit.runner.worker/DiscoverUtil.cs

```diff
@@ -33,19 +51,23 @@ protected override bool Visit(ITestCaseDiscoveryMessage testCaseDiscovered)
                     testCaseDiscovered.TestAssembly.Assembly.AssemblyPath,
                     testCase.Traits);
 
-                Console.WriteLine(testCase.DisplayName);
                 _writer.Write(TestDataKind.Value);
                 _writer.Write(testCaseData);
 
                 return _writer.IsConnected;
             }
+
+            public void Dispose()
+            {
+                ((IDisposable)Finished).Dispose();
+            }
         }
 
         internal static void Go(string fileName, Stream stream)
         {
             using (AssemblyHelper.SubscribeResolve())
             using (var xunit = new XunitFrontController(
-                AppDomainSupport.IfAvailable,
+                AppDomainSupport.Denied,
```

Comment:
This is definitely a problem. I wouldn't expect it to be difficult though. The worker process code is in https://github.com/Pilchie/xunit.runner.wpf/blob/master/xunit.runner.wpf/Impl/RemoteTestUtil.cs